src/Logging/ServiceCollectionExtensions.cs

```diff
@@ -50,14 +53,20 @@ public static IServiceCollection AddOmexLogging(this IServiceCollection serviceC
 			serviceCollection.TryAddTransient<IExecutionContext, BaseExecutionContext>();
 			serviceCollection.TryAddTransient<IExternalScopeProvider, LoggerExternalScopeProvider>();
 
+			AddOmexLegacyLogging(serviceCollection);
```

Comment:
What are you achieving my extracting code into a separate function?

Reply:
I know some consumers are using this AddOmexLogging directly. I make this obsolete to explicitly remind them not to use this method.
If they do not use this method, there is still a need for EmptyServiceContext, BaseExecutionContext, LoggerExternalScopeProvider. That is why I copy them to [HostBuilderExtensions.cs](https://github.com/microsoft/Omex/pull/673/files#diff-4bf858f6b9cd29174b972054a6df039db4dcb0c3f43c4316e5e3e19eff5b7f2d)

Reply:
I don't get it. You extracted some code into AddOmexLegacyLogging which is a private method and cannot be called from anywhere else. Why?

Reply:
Oh, I see the confusion. I revert and remove ~AddOmexLegacyLogging~. 
1. The goal 1 here is to mark AddOmexLogging [Obsolete], and delete it in the future.

2. Goal 2 is to still keep EmptyServiceContext, BaseExecutionContext, and LoggerExternalScopeProvider registered. For goal 2, I copied them to  `AddOmexServiceFabricDependencies()`.  This is a downstream place to catch-all scenario when consumers follow these call orders:
a. BuildStatelessService -> BuildServiceFabricService -> AddOmexServiceFabricDependencies
b. BuildStatefulService -> BuildServiceFabricService -> AddOmexServiceFabricDependencies

For Omex consumers, we all called BuildStatelessService / BuildStatefulService  which in turns call AddOmexServiceFabricDependencies and will get EmptyServiceContext, BaseExecutionContext, and LoggerExternalScopeProvider registered

Reply:
The both EmptyServiceContext  and OmexServiceFabricContext both implement IServiceContext.
EmptyServiceContext is used when consumers only use AddOmexLogging() but dont use BuildServiceFabricService.
OmexServiceFabricContext is used when  consumers use BuildServiceFabricService.

Reply:
Therefore, EmptyServiceContext and LoggerExternalScopeProvider dont need to be copied to AddOmexServiceFabricDependencies

Reply:
Kudo. I see your point. If they call AddOmexLogging first, the IServiceContext will be resolved into EmptyServiceContext.  
Subsequently, if they call any of BuildStatelessService or BuildServiceFabricService or AddOmexServiceFabricDependencies, IServiceContext will **not** be resolved in OmexServiceFabricContext.

In other words, if Omex consumers are explicitly use AddOmexLogging **and** BuildStatelessService() , they will lose SFContext. 

Is my understanding correct?

By the way, this is very good to spot Artem. It should be discussed out of this PR.